wordpress-mobile/WordPress-Android
# [Reader Improvements] Implement new design for "Sites to follow" section

### Do not merge
Should only be merged after https://github.com/wordpress-mobile/WordPress-Android/pull/19249

Part of #19083 

This PR updates the follow/unfollow button design from the "Sites to follow" view and also creates a new layout file to be used when the new design is implemented on "Sites to follow".

Design specs: zQnohyMpLzBzQ5jzMMKni3-fi

To test:
1 - Run JP and sign in;
2 - Open debug settings and enable `ReaderImprovementsFeatureConfig` FF;
3 - Open reader;
4 - Select "Discover" tab;
5 - Scroll until you find the "Sites to follow" section;
6 - Verify the layout: the section should have the new design. Make sure that the follow/unfollow action still works as expected;
7 - Repeat step 2 but this time disable the FF;
8 - Repeat steps 3-5 and verify the layout: the section layout should appear like it currently does in prod;

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
Changes were only made to views

PR submission checklist:


UI Changes testing checklist:
